lib/dynamo.js

```diff
@@ -0,0 +1,69 @@
+const log = require('lambda-log')
+const AWS = require('aws-sdk')
+const { ArrangementInvalidError, DynamodbGetError, MissingEnvError } = require('./errors')
+
+const region = process.env.ARRANGEMENTS_DDB_REGION || 'us-east-1'
+const clientOptions = { region, maxRetries: 5, httpOptions: { timeout: 1000 } }
+
+// TODO: shouldn't this happen automagically, with clientOptions.region set?
+const endpoint = new AWS.Endpoint(`sts.${region}.amazonaws.com`)
+const sts = new AWS.STS({ endpoint })
+
+// optionally load a client using a role
+exports.client = async () => {
+  if (process.env.ARRANGEMENTS_DDB_ACCESS_ROLE) {
+    const RoleArn = process.env.ARRANGEMENTS_DDB_ACCESS_ROLE
+    const RoleSessionName = 'dovetail-counts-lambda-dynamodb'
+    try {
+      const data = await sts.assumeRole({ RoleArn, RoleSessionName }).promise()
+      const { AccessKeyId, SecretAccessKey, SessionToken } = data.Credentials
+      return new AWS.DynamoDB({
+        ...clientOptions,
+        accessKeyId: AccessKeyId,
+        secretAccessKey: SecretAccessKey,
+        sessionToken: SessionToken,
+      })
+    } catch (err) {
+      log.error(err, { msg: 'sts error', role: RoleArn })
+      return new AWS.DynamoDB(clientOptions)
+    }
+  } else {
+    return new AWS.DynamoDB(clientOptions)
+  }
+}
+
+/**
+ * Actual getItem call
+ */
+exports._getItem = async (params, client = null) => {
+  client = client || (await exports.client())
+  return client.getItem(params).promise()
+}
+
+/**
+ * Get an arrangement json
+ */
+exports.getArrangement = async (digest, client = null) => {
+  if (!process.env.ARRANGEMENTS_DDB_TABLE) {
+    throw new MissingEnvError('You must provide a ARRANGEMENTS_DDB_TABLE')
+  }
+
+  const params = { Key: { digest: { S: digest } }, TableName: process.env.ARRANGEMENTS_DDB_TABLE }
+
+  let result = null
+  try {
+    result = await exports._getItem(params, client)
+  } catch (err) {
+    throw new DynamodbGetError(`Failed to get ${digest}`, err)
+  }
+
+  if (result.Item && result.Item.data && result.Item.data.S) {
```

Comment:
This is on node 14 I think, so could be `result?.Item?.data?.S`

Reply:
Hrmm, the typescript linter plugin in Atom hates that, even though it's legal.  Maybe I need to upgrade something...

Reply:
Actually, I can't use that operator because it breaks the tests?  Since there is a `jest.genMockFromModule('../dynamo')` ... and Jest currently only supports mocking vanilla javascript files.

Reply:
Ok, grandpa, have fun in 2019